src/ngraph/op/cum_sum.hpp

```diff
@@ -0,0 +1,99 @@
+       ///
+       /// |                      | Description                                        |
+       /// | -------------------- | -------------------------------------------------- |
+       /// | `reverse`            | if set to 1, performs the sum in reverse direction |
+       ///
+       /// ## Inputs
+       ///
+       /// |       | Description                                            |
+       /// | ----- | ------------------------------------------------------ |
+       /// | `arg` | An input tensor of any shape and numeric element type. |
+       ///
+       /// |       | Description                                                     |
+       /// | ----- | --------------------------------------------------------------- |
+       /// | `axis`| An axis position along which cumulative sum must be performed.  |
+       ///
+       /// ## Output
+       ///
+       /// | Description                                                                         |
+       /// | ------------------------------------------------------------------------------------|
+       /// | Output tensor of the same type as `arg` with cumulative sums of the arg's elements  |
+       //
+       // clang-format off
+       class CumSum : public Op 
+       {
+       public:
+           NGRAPH_API
+           static constexpr NodeTypeInfo type_info{ "CumSum", 0 };
+           const NodeTypeInfo& get_type_info() const override { return type_info; }
+           /// \brief Constructs a cumulative summation operation.
+           CumSum() = default;
+          
+           /// \brief Constructs a cumulative summation operation.
+           ///
+           /// \param arg The tensor to be summed.
+           /// \param axis The axis position along which cumulative sum must be performed.
+           CumSum(const Output<Node>& arg, const int64_t axis = 0, const bool exclusive = false, const bool reverse = false); 
+    
+           /// \brief Constructs a cumulative summation operation.
+           ///
+           /// \param arg The tensor to be summed.
+           /// \param axis The axis position along which cumulative sum must be performed.
+           CumSum(const Output<Node>& arg, const Output<Node>& axis, const bool exclusive = false, const bool reverse = false);
+
+           virtual std::shared_ptr<Node>
+               copy_with_new_args(const NodeVector& new_args) const override;
+
+           /// \return The default value for CumSum.
+           virtual std::shared_ptr<Node> get_default_value() const override;
+           int64_t get_axis() const { return m_axis; }
+           bool is_exclusive() const { return m_exclusive; }
+           bool is_reverse() const { return m_reverse; }
+       protected:
+           virtual void generate_adjoints(autodiff::Adjoints& adjoints,
+               const NodeVector& deltas) override;
+    
+       private:
+	   bool m_exclusive;
+	   bool m_reverse;
+	   int64_t m_axis;
```

Comment:
In order to support the ONNX version of `CumSum`, we will need `axis` to be a dynamic input, not a static attribute of the op.